### PR TITLE
Remove `CholeskyGrad` Op

### LINKED ISSUE #1098: Remove `CholeskyGrad` Op
I'm wondering if we should remove the [CholeskyGrad](https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/slinalg.py#L126-L190) Op entirely because we have a working [L_Op](https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/slinalg.py#L70) inside the `Cholesky` Op and are no longer using `CholeskyGrad` Op.

Thanks to @ricardoV94 for sharing this reference (https://github.com/aesara-devs/aesara/commit/1c8f8d6ace92683359ef74401e57a7b09a209f7e), which shows that `CholeskyGrad` was used in the original implementation of `Cholesky` but was later removed.

### PR DESCRIPTION
**Pull request by sudarsan2k5**
Fixes #1098 

This PR removes `CholeskyGrad` Op and all the tests related to it.

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [x] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [x] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [x] There are tests covering the changes introduced in the PR.

### CODE REVIEW

**Review by brandonwillard — APPROVED** [review]

**Comment by brandonwillard** [comment]
We have to squash these before/when we merge; otherwise, this looks good, thanks!

**Comment by rlouf** [comment]
I just rebased your changes on `main`, will merge if the tests pass. Thank you for contributing!

**Review by rlouf — APPROVED** [review]

**Pull request merged**
